AE-961: Event Monitor condition check and tests (#6)
arbitraryexecution/forta-agent-templates: admin-events/SETUP.md

1. Open the `agent-config.json` file.

2. `developerAbbreviation` (required) - Type in your desired abbreviation to specify your name or your development
team name.  For example, Arbitrary Execution uses the abbreviation `"AE"` for its `developerAbbreviation` value.

3. `protocolName` (required) - Type in the name of the protocol.  For example, for the Uniswap protocol you may
type in `"Uniswap"` or `"Uniswap V3"`, for the SushiSwap protocol you may type in `"Sushi"` or `"SushiSwap"`, etc.

4. `protocolAbbreviation` (required) - Type in an appropriate abbreviation for the value in `protocolName`.  For
example, `"Uniswap"` may be abbreviated `"UNI"` and `"SushiSwap"` may be abbreviated `"SUSH"`, etc.

5.  `contracts` (required) - The Object value for this key corresponds to contracts that we want to monitor events
for.  Each key in the Object is a contract name that we can specify, where that name is simply a string that we use
as a label when referring to the contract (the string can be any valid string that we choose, it will not affect the
monitoring by the agent).  The Object corresponding to each contract name requires an address key/value pair, abi

> file key/value pair, and either an `events` key or a `proxy` key.  For the case of an `events` key, the corresponding

Reviewer: ```suggestion
file key/value pair, and an `events` key, `proxy` key, or both.  For the case of an `events` key, the corresponding
```

      "events": {
        "NewAdmin": {
          "type": "Suspicious",
          "severity": "Medium"
        }
      }
    } 
  }
```

If we wanted to add an expression to check that the address of `newAdmin` emitted by the `NewAdmin` event is not the zero address,
the config would look like the following:

```json
  "contracts": {
    "GovernorBravo": {
      "address": "0x408ED6354d4973f66138C91495F2f2FCbd8724C3",
      "abiFile": "GovernorBravo.json",
      "events": {
        "NewAdmin": {
          "expression": "newAdmin !== 0x0000000000000000000000000000000000000000",
          "type": "Suspicious",
          "severity": "Medium"
        }
      }
    }
  }
```

Note that any unused entries in the configuration file must be deleted for the agent to work.  The original version
of the configuration file contains several placeholders to show the structure of the file, but these are not valid
entries for running the agent.

6. If a contract is a proxy for another contract, where events will be emitted as if they are coming from the proxy
instead of from the underlying implementation contract, the entry in the `agent-config.json` file may look like the
following:

```json

>   "adminEvents": {

Reviewer 2: this should be `"contracts"`

      ContractABIFileN.json
```

10. Install all related `npm` packages using `npm i`.  This will create a `package-lock.json` file alongside
package.json.

11. Once the `agent-config.json` file is populated and all corresponding ABI files are in the correct locations
referred to in the `agent-config.json` file, the agent is complete.  Please test the agent against transactions
that contain events that should trigger the agent.  Please also test the agent against transactions that should

> not trigger the agent.  Although not provided here, please create tests that will verify the functionality of
> the agent code for positive cases, negative cases, and edge cases (e.g. when errors occur).

Reviewer: ```suggestion
not trigger the agent.
```
